firedrake/solving_utils.py

```diff
@@ -165,7 +167,8 @@ class _SNESContext(object):
     Firedrake level information.
     """
     @PETSc.Log.EventDecorator()
-    def __init__(self, problem, mat_type, pmat_type, appctx=None,
+    def __init__(self, problem, mat_type, pmat_type,
+                 sub_mat_type=None, appctx=None,
```

Comment:
Small thing but a more intuitive parameter ordering might be `mat_type`, `sub_mat_type` then `pmat_type`.

Reply:
I also added a ``. The new order is `mat_type, pmat_type, sub_mat_type, sub_pmat_type`.